dockeradapter/filterdockerstream.go

```diff
@@ -24,62 +24,98 @@ const (
 	stdWriterSizeIndex = 4
 )
 
+// frameHeader holds the stream type and payload size for a Docker stream frame
+type frameHeader struct {
+	streamType  int
+	payloadSize int
+}
+
 // FilterDockerStream reads the requested stream from *bufio.Reader, strips the
-// prefix headers added by Dockers stdWriter.Write(p []byte) and returns a slice
+// frameHeader added by Dockers stdWriter.Write(p []byte) and returns a slice
 // of strings for each line received (splitting on '/n').
 //
 // If the stream processes without error but no lines are found then it will
 // return an empty slice of strings.
 //
 // The following streamTypes are supported (from the Docker stdcopy package):
 //  - Stdin    - 0
-//  - Stdout   - 1
+//  - Stderr   - 1
```

Comment:
Thanks, this was a hangover from changing the logging to Stdout while debugging.. we need to replace the current stopgap global logger with an interface we can pass around instead - would make debugging/testing a lot easier as well.